src/components/data/DataTable.tsx

```diff
@@ -305,26 +311,40 @@ export default function DataTable<Row>({
       <TableRow
         key={idx}
         selected={selection.includes(row)}
-        onClick={e => selectRow(row, e.shiftKey ? 'extend' : 'replace')}
+        onClick={e => {
+          selectRow(row, e.shiftKey ? 'extend' : 'replace');
+          onClickRow?.(row);
+        }}
         onDblClick={() => confirmRow(row)}
         onKeyDown={event => handleKeyDown(event, row)}
+        elementRef={el => {
+          el?.addEventListener('select-row', e => {
+            selectRow(
+              row,
+              (e as CustomEvent).detail.shiftKey ? 'extend' : 'replace',
+            );
+          });
```

Comment:
I inlined this here just for simplicity. If we want to move forward, we would have to add proper refs, and attach/remove event listeners to them via `useEffect`.